test/parallel/test-timers-clearImmediate-als.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+const common = require('../common');
+const { AsyncLocalStorage } = require('async_hooks');
+
+// This is an asynclocalstorage variant of test-timers-clearImmediate.js
+const asyncLocalStorage = new AsyncLocalStorage();
+const N = 3;
+
+function next() {
+  const fn = common.mustCall(onImmediate);
+  asyncLocalStorage.run(new Map(), common.mustCall(() => {
+    const immediate = setImmediate(fn);
+    const store = asyncLocalStorage.getStore();
+    store.set('immediate', immediate);
+  }));
+}
+
+function onImmediate() {
+  const store = asyncLocalStorage.getStore();
+  const immediate = store.get('immediate');
```

Comment:
Given that `clearImmediate(undefined)` and `clearImmediate(null)` do not throw, it may be a good idea to assert that `immediate` here is the expected value.

Reply:
@jasnell, thanks. Asserted to check `immediate`. PTAL.